### PR TITLE
Add missing tag_map functions to idempotent branch

### PR DESCRIPTION
**Pull request by pgnickb**

### CODE REVIEW

File: migration/idempotent/008-tracing-functions.sql
Comment (jgpruitt) [inline migration/idempotent/008-tracing-functions.sql:639]: Is it possible to check here if the tag_map type currently has a subscript handler attached and alter the type to attach it if it is missing? If it's possible, I think that would address the case in which a user installs on v13 and then upgrades to v14 and is missing the subscript functionality.

**Review by jgpruitt — CHANGES_REQUESTED** [review]

**Review by cevian — APPROVED** [review]

File: migration/idempotent/008-tracing-functions.sql
Comment (cevian) [reply migration/idempotent/008-tracing-functions.sql:639]: Does that need to hold up the PR? seems like a post-GA thing?

Comment (jgpruitt) [reply migration/idempotent/008-tracing-functions.sql:641]: nope. doesn't need to hold it up, but if it's super quick might as well?

File: migration/idempotent/008-tracing-functions.sql
Comment (jgpruitt) [inline migration/idempotent/008-tracing-functions.sql:640]: check the `typsubscript` column of the `pg_type` table for the `tag_map` type, and only alter the type if the subscript is missing

**Review by jgpruitt — CHANGES_REQUESTED** [review]

File: migration/idempotent/008-tracing-functions.sql
Comment (pgnickb) [reply migration/idempotent/008-tracing-functions.sql:633]: This was a deliberate choice over $$, to make the statement more readable in the logs (it looks as a oneliner in the log output), but I will replace it with a regular statement, as there is no real need for EXECUTE here alltogether.

File: migration/idempotent/008-tracing-functions.sql
Comment (pgnickb) [reply migration/idempotent/008-tracing-functions.sql:640]: I considered this but didn't come up with a good reason for this. Will do

File: migration/incremental/015-tracing-redesign.sql
Comment (cevian) [inline migration/incremental/015-tracing-redesign.sql:307]: But this shouldn't be needed due to https://github.com/timescale/promscale_extension/blob/develop/migration/bootstrap/000-migration-table.sql#L2
```suggestion
    SET search_path = pg_catalog, pg_temp;
```

File: migration/incremental/015-tracing-redesign.sql
Comment (cevian) [inline migration/incremental/015-tracing-redesign.sql:378]: same as one get rid of this

File: migration/idempotent/008-tracing-functions.sql
Comment (cevian) [inline migration/idempotent/008-tracing-functions.sql]:
```suggestion
                typsubscript != '_ps_trace.tag_v_subscript_handler'::regproc
```

File: migration/idempotent/008-tracing-functions.sql
Comment (cevian) [inline migration/idempotent/008-tracing-functions.sql]:
```suggestion
                typsubscript != 'ps_trace.tag_map_subscript_handler'::regproc
```

Comment (jgpruitt) [inline migration/idempotent/008-tracing-functions.sql:640]: This will alter the type if the type already has the subscript, right? Don't we want to alter the type if the type DOES NOT have the subscript? I think we should check if the typsubscript is null or something, right?

**Review by jgpruitt — CHANGES_REQUESTED** [review]

File: migration/idempotent/008-tracing-functions.sql
Comment (cevian) [inline migration/idempotent/008-tracing-functions.sql:645]:
```suggestion
                --this is needed upgrade pg13->14
                ALTER TYPE ps_trace.tag_map SET (SUBSCRIPT = ps_trace.tag_map_subscript_handler);
```

**Review by cevian — CHANGES_REQUESTED** [review]

**Review by cevian — APPROVED** [review]

File: migration/idempotent/008-tracing-functions.sql
Comment (pgnickb) [reply migration/idempotent/008-tracing-functions.sql]: Yes, but my assumption was that this is essentially a no-op if it already has one, so I didn't bother with the check. Fixed now.

**Review by jgpruitt — APPROVED** [review]

File: migration/incremental/015-tracing-redesign.sql
Comment (JamesGuthrie) [inline migration/incremental/015-tracing-redesign.sql:314]: Why is this type in `ps_trace`, but `tag_v` is in `_ps_trace`? Was that on purpose, or did it just happen?

**Review by JamesGuthrie — APPROVED** [review]

**Comment by JamesGuthrie** [comment]
Also: can you squash the commits down a bit before merging?

**Comment by pgnickb** [comment]
Yeah we need to make sure we squash and merge this one. It is rather simple anyway.

**Pull request merged**
